ivy/src/main/scala/sbt/IvyScala.scala

```diff
@@ -47,17 +50,39 @@ private object IvyScala {
   /** Performs checks/adds filters on Scala dependencies (if enabled in IvyScala). */
   def checkModule(module: DefaultModuleDescriptor, conf: String, log: Logger)(check: IvyScala): Unit = {
     if (check.checkExplicit)
-      checkDependencies(module, check.scalaBinaryVersion, check.configurations, log)
+      checkDependencies(module, check.scalaOrganization, check.scalaBinaryVersion, check.configurations, log)
     if (check.filterImplicit)
       excludeScalaJars(module, check.configurations)
-    if (check.overrideScalaVersion)
-      overrideScalaVersion(module, check.scalaFullVersion)
+    if (true || check.overrideScalaVersion)
+      overrideScalaVersion(module, check.scalaOrganization, check.scalaFullVersion)
   }
-  def overrideScalaVersion(module: DefaultModuleDescriptor, version: String): Unit = {
-    overrideVersion(module, Organization, LibraryID, version)
-    overrideVersion(module, Organization, CompilerID, version)
-    overrideVersion(module, Organization, ReflectID, version)
+
+  class OverrideScalaMediator(scalaOrganization: String, scalaVersion: String) extends DependencyDescriptorMediator {
```

Comment:
Could this and perhaps the other new methods here be unit tested under https://github.com/sbt/sbt/tree/0.13/ivy/src/test/scala?